insights/Extract_MC_Insights_DBX.py

```diff
@@ -0,0 +1,73 @@
+# Databricks notebook source
+# MAGIC %md # Environment Setup
+
+# COMMAND ----------
+
+#Only run this if you have not installed pycarlo on the DBX cluster already
+#pip install pycarlo
+
+# COMMAND ----------
+
+# DBTITLE 1,Runtime Variables
+# Preferred: Monte Carlo Credentials stored in DBX Secret Key Repo called "monte-carlo-creds":
+#mcd_id = dbutils.secrets.get(scope = "monte-carlo-creds", key = "mc-key-id")
+#mcd_token = dbutils.secrets.get(scope = "monte-carlo-creds", key = "mc-secret-id")
+
+#Otherwise hardcode for testing purposes only!
+mcd_id = 'your-id-here'
+mcd_token = 'your-secret-here'
+
+# Full List of Reports Needed, Edit as Needed for Databricks
+insight_names = ["monitors","cleanup_suggestions","events","incident_history"]
+insight_report_names = ["monitors.csv","cleanup_suggestions.csv","events.csv","incident_history.csv"]
+
+# Other variables which you should customize as needed:
+mcd_profile=""
+table_schema = 'your_destination_schema_name'
+user = 'your_user_name_in_DBFS'
+
+# COMMAND ----------
+
+# MAGIC %md # Load Insights to DBX
+
+# COMMAND ----------
+
+from pycarlo.core import Client, Query, Mutation, Session
+import requests
+from pyspark.sql.functions import *
+import io
+import pandas as pd
+from datetime import *
+
+client = Client(session=Session(mcd_id=mcd_id, mcd_token=mcd_token,mcd_profile=mcd_profile))
+today = datetime.today()
+
+for i in range(len(insight_report_names)):
+    print("Looking for Insight Report: {}".format(insight_report_names[i]))
+    query=Query()
+    query.get_report_url(insight_name=insight_names[i],report_name=insight_report_names[i]).__fields__('url')
+    report_url=client(query).get_report_url.url
+    r = requests.get(report_url).content
+    
+    # Customize the naming scheme of the loaded tables here:
+    table_name = "mc_src_"+insight_names[i]
+    filename = insight_report_names[i]
+    
+    df=pd.read_csv(io.StringIO(r.decode('utf-8')))  
+    #display(df) #Uncomment to see the data before it is loaded to a table
+    
+    #changing column spaces to underscores (if there are any)
+    df.columns = df.columns.str.replace(' ','_')
+    print('Creating Spark Data Frame')
+    DF = spark.createDataFrame(df).withColumn("load_date", lit(date(today.year, today.month, today.day)))
+    dbutils.fs.rm("dbfs:/FileStore/shared_uploads/{}/{}".format(table_name,user), True)  
+    DF.write.format("delta").mode("overwrite").option("mergeSchema", "true").save("dbfs:/FileStore/shared_uploads/{}/{}".format(user,table_name))  
+    spark.sql("DROP TABLE if exists {}.{}".format(table_schema,table_name))  
+    spark.sql("CREATE TABLE {}.{} USING DELTA LOCATION 'dbfs:/FileStore/shared_uploads/{}/{}'".format(table_schema,table_name,user,table_name))  
```

Comment:
Databricks recommends using REPLACE instead of dropping and re-creating delta tables

https://docs.databricks.com/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html#parameters

Reply:
I'd also recommend leaving a comment at the top of this notebook that this script deletes resources. 

Just so folks are aware. Thanks!

Reply:
I simplified the script to load direct to the existing schema location w/o adding/deleting files on dbfs or removing the table.  Take a look at the new copy.